### PR TITLE
Report (virtual/simulated) time for TX receive from reference server

### PR DESCRIPTION
**Pull request by MarcFontaine**
Extend the reference TX submission server to report the time of TX reception when running
in IOSim (with virtual time)

### CODE REVIEW

File: tx-generator-shelley/test/Cardano/Benchmarking/ReferenceServer.hs [hunk 1]
```diff
@@ -194,22 +195,22 @@ initialServerState = ServerState 0 Seq.empty Map.empty Map.empty 0
...
-     (Ord txid, Show txid, Show tx, Monad m)
+  :: forall txid tx m out.
+     (out ~ [ (tx,UTCTime) ], Ord txid, Show txid, Show tx, MonadTime m)
   => Tracer m (TraceEventServer txid tx)
   -> (tx -> txid)
   -> Word16  -- ^ Maximum number of unacknowledged txids
   -> Word16  -- ^ Maximum number of txids to request in any one go
   -> Word16  -- ^ Maximum number of txs to request in any one go
-  -> TxSubmissionServerPipelined txid tx m [tx]
+  -> TxSubmissionServerPipelined txid tx m out
```
Comment (CodiePP) [inline tx-generator-shelley/test/Cardano/Benchmarking/ReferenceServer.hs:205]: it's not enough to put the type `[(tx,UTCTime)]` here?

**Review by CodiePP — APPROVED** [review]
just a question about the type restriction

File: tx-generator-shelley/test/Cardano/Benchmarking/ReferenceServer.hs (hunk 1 above)
Comment (MarcFontaine) [reply tx-generator-shelley/test/Cardano/Benchmarking/ReferenceServer.hs:205]: That would have worked as well.

Comment (MarcFontaine) [reply tx-generator-shelley/test/Cardano/Benchmarking/ReferenceServer.hs:205]: The same type `[(tx,UTCTime)]` is used in about 7 or 8 places, I thought it was better to name it as `out`. Before the change it was `[tx]` but when changing it I was not sure if every `[tx]` will be replaced with `out`.

**Pull request merged**